nanomq/bridge.c

```diff
@@ -127,6 +127,43 @@ create_disconnect_msg()
 	return msg;
 }
 
+int
+bridge_handle_sub_reflection(nano_work *work, conf_bridge *bridge)
+{
+	for (size_t t = 0; t < bridge->count; t++) {
+		conf_bridge_node *node = bridge->nodes[t];
+		if (node->enable) {
+			for (size_t i = 0; i < node->sub_count; i++) {
+				char *topic = work->pub_packet->var_header.publish.topic_name.body;
+				int len = work->pub_packet->var_header.publish.topic_name.len;
+				if (topic != NULL && node->sub_list[i]->remote_topic != NULL) {
+					if (strncmp(topic, node->sub_list[i]->remote_topic, len) == 0) {
```

Comment:
What if the topic is the prefix of the remote_topic.
My suggestion. Use strcmp rather than strncmp.